contract/src/offer-up.contract.js

```diff
@@ -90,36 +81,47 @@ export const start = async zcf => {
    * The `Items` amount must use the `Item` brand and a bag value.
    */
   const proposalShape = harden({
-    give: { Price: M.gte(tradePrice) },
-    want: { Items: { brand: itemBrand, value: M.bag() } },
+    give: { Price: M.eq(subscriptionPrice) },
+    want: { Items: { brand: M.any(), value: M.bag() } },
     exit: M.any(),
   });
 
   /** a seat for allocating proceeds of sales */
   const proceeds = zcf.makeEmptySeatKit().zcfSeat;
 
+  const subscriptions = new Map();
+
   /** @type {OfferHandler} */
-  const tradeHandler = buyerSeat => {
-    // give and want are guaranteed by Zoe to match proposalShape
-    const { want } = buyerSeat.getProposal();
+  const tradeHandler = async (buyerSeat, offerArgs) => {
+    // @ts-ignore
+    const userAddress = offerArgs.userAddress;
```

Comment:
what's the goal here?
This address is forgeable.
Can you key the map on seats instead of addresses?

Reply:
see [suggested alternative](r1746072966) below